.github/workflows/build-icu.yml

```diff
@@ -59,14 +59,194 @@ jobs:
+        with:
+          servers: |
+            [{
+                "id": "github",
+                "username": "${{ github.actor }}",
+                "password": "${{ github.token }}"
+            },{
+                "id": "githubicu",
+                "username": "${{ github.actor }}",
+                "password": "${{ github.token }}"
+            },{
+                "id": "githubcldr",
+                "username": "${{ github.actor }}",
+                "password": "${{ github.token }}"
+            }]
+      - name: Create empty CLDR dir, unpack cldr-prod
+        run: mkdir -p cldr cldr-prod && cd cldr-prod && unzip ../cldr-prod.zip
+      # Don't actually need CLDR
+      # - name: Clone CLDR
+      #   uses: actions/checkout@v2
+      #   with:
+      #     ref: ${{ github.event.inputs.cldr-ref }}
+      #     lfs: false
+      #     repository: ${{ github.event.inputs.cldr-repo }}
+      #     path: cldr
+      - name: Clone ICU
+        uses: actions/checkout@v2
+        with:
+          repository: ${{ github.event.inputs.icu-repo }}
+          ref: ${{ github.event.inputs.icu-ref }}
+          lfs: false # not needed here
+          path: icu
+      - name: Cache local Maven repository
+        uses: actions/cache@v2
+        with:
+          path: ~/.m2/repository
+          key: ${{ runner.os }}-maven-${{ hashFiles('icu/tools/**/pom.xml') }}
+          restore-keys: |
+            ${{ runner.os }}-maven-
+      - name: Install CLDR tools
+        # TODO: Should not have to do this, see ICU-21425
+        # Do this after restoring cache.
+        run: >
+          mvn -B install:install-file   -Dproject.parent.relativePath=""   -DgroupId=org.unicode.cldr
+          -DartifactId=cldr-api   -Dversion=0.1-SNAPSHOT   -Dpackaging=jar   -DgeneratePom=true
+          -Dfile=cldr-code/cldr-code.jar
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      - name: Generate ICU Data
+        run: >
+          env
+          TOOLS_ROOT=$(pwd)/icu/tools/
+          CLDR_DIR=$(pwd)/cldr
+          CLDR_DATA_DIR=$(pwd)/cldr-prod/
+          ant -f icu/tools/cldr/cldr-to-icu/build-icu-data.xml clean all && ln icu/icu4c/LICENSE icu/icu4c/source/data/
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      - name: Zip up icu-data.zip
+        run: 'cd icu/icu4c/source && zip -rlq ../../../icu-data.zip data'
+      - name: Upload icu-data.zip
+        uses: actions/upload-artifact@v2
+        with:
+          name: icu-data
+          path: icu-data.zip
+  # clang release build with some options to enforce useful constraints.
+  # Includes dependency checker on an in-source, optimized build.
+  # Includes checking @draft etc. API tags vs. ifndef guards like
+  # U_HIDE_DRAFT_API and U_FORCE_HIDE_DRAFT_API.
+  # (FORCE guards make this tool pass but won't compile to working code.
+  # See the testtagsguards.sh script for details.)
+  icu4c-clang:
+    runs-on: ubuntu-latest
```

Comment:
Wonder if maybe giving it a name, perhaps something like (?):
`    name: Build and test ICU4C, with new data`